mixer/pkg/direct/envoymetrics/server.go

```diff
@@ -0,0 +1,386 @@
+package envoymetrics
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	mpb "github.com/envoyproxy/go-control-plane/envoy/service/metrics/v2"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	mixerpb "istio.io/api/mixer/v1"
+	prometheus "istio.io/gogo-genproto/prometheus"
+	"istio.io/istio/pkg/log"
+)
+
+// TODO: support for buckets and quantiles is really really poor. Need a better story around that information.
+
+const (
+	serviceNodeSeparator = "~"
+	kubePrefix           = "kubernetes://"
+	protocol             = "envoy_metrics_service"
+
+	// attribute names
+	destinationIP   = "destination.ip"
+	destinationUID  = "destination.uid"
+	contextProtocol = "context.protocol"
+	nodeID          = "envoy.api.v2.core.node.id"
+	cluster         = "envoy.api.v2.core.node.cluster"
+	metadata        = "envoy.api.v2.core.node.metadata"
+	buildVersion    = "envoy.api.v2.core.node.build_version"
+	region          = "envoy.api.v2.core.node.locality.region"
+	zone            = "envoy.api.v2.core.node.locality.zone"
+	subzone         = "envoy.api.v2.core.node.locality.subzone"
+	familyName      = "io.prometheus.client.metricfamily.name"
+	familyType      = "io.prometheus.client.metricfamily.kind" // use kind here, because 'type' will cause issues with the frontend for the expression lang
+	metricLabels    = "io.prometheus.client.metric.labels"
+	counterVal      = "io.prometheus.client.metric.counter.value"
+	gaugeVal        = "io.prometheus.client.metric.gauge.value"
+	summaryCount    = "io.prometheus.client.metric.summary.sample_count"
+	summarySum      = "io.prometheus.client.metric.summary.sample_sum"
+	quantiles       = "io.prometheus.client.metric.summary.quantiles"
+	untypedVal      = "io.prometheus.client.metric.untyped.value"
+	histogramCount  = "io.prometheus.client.metric.histogram.sample_count"
+	histogramSum    = "io.prometheus.client.metric.histogram.sample_sum"
+	buckets         = "io.prometheus.client.metric.histogram.buckets"
+	metricTimestamp = "io.prometheus.client.metric.timestamp"
+)
```

Comment:
I hope you plan to import all these constants from their packages instead of defining them here?